core/src/com/unciv/logic/map/InfluenceMap.kt

```diff
@@ -0,0 +1,206 @@
+        for (tile in viewable) {
+            tile.enemyInfluenceScore = 0.0F
+            tile.friendlyInfluenceScore = 0.0F
+        }
+        val tilesWithEnemies = (tilesToCheck ?: viewable)
+                .filter { containsEnemyUnit(it, viewingCiv) }
+        val tilesWithNeutralCombatant = (tilesToCheck ?: viewable)
+                .filter { containsNeutralUnit(it, viewingCiv) }
+        val tilesWithFriendlyCombatant = (tilesToCheck ?: viewable)
+                .filter { containsFriendlyUnit(it, viewingCiv) }
+        setEnemyTiles(tilesWithEnemies)
+        setNeutralTiles(tilesWithNeutralCombatant)
+        setFriendlyTiles(tilesWithFriendlyCombatant)
+
+    }
+    
+    private fun setEnemyTiles(enemyTiles: List<TileInfo>) {
+        for (enemyTile in enemyTiles) {
+            for (unit in enemyTile.getUnits()) {
+                if (unit.type.isMilitary()) {
+                    enemyTile.enemyInfluenceScore += (unit.baseUnit.strength * (1 + enemyTile.getDefensiveBonus())) * (unit.health/100.0F)
+                    if (unit.type.isMelee()) {
+                        val tilesInMoveRange = unit.movement.getDistanceToTilesWithinTurn(
+                            unit.getTile().position,
+                            unit.getMaxMovement().toFloat()
+                        ).keys.asSequence().filter { it.position != enemyTile.position }
+                        for (tile in tilesInMoveRange) {
+                            tile.enemyInfluenceScore += (unit.baseUnit.strength * (1 + tile.getDefensiveBonus())) * (unit.health / 100.0F)
+                        }
+                    }
+                    if (unit.type.isRanged()) {
+                        val rangeOfAttack = unit.getRange()
+                        val tilesInTargetRange =
+                                if (unit.hasUnique("Ranged attacks may be performed over obstacles") || unit.type.isAirUnit())
+                                    enemyTile.getTilesInDistance(rangeOfAttack)
+                                else enemyTile.getViewableTilesList(rangeOfAttack)
+                                        .asSequence()
+                        for (tile in tilesInTargetRange) {
+                            tile.enemyInfluenceScore += (unit.baseUnit.rangedStrength * (1 + tile.getDefensiveBonus())) * (unit.health / 100.0F)
+                        }
+                    }
+                } else if (unit.name == Constants.greatGeneral || unit.baseUnit.replaces == Constants.greatGeneral) {
+                    val tilesInAffectedRange = enemyTile.getTilesInDistance(2)
+                    for (tile in tilesInAffectedRange) {
+                        tile.enemyInfluenceScore *= 1.15F
+                    }
+                } else if (unit.type.isCivilian()) {
+                    enemyTile.enemyInfluenceScore = -2.0F
+                }
+            }
+            if (enemyTile.isCityCenter()) {
+                val enemyCity = enemyTile.getCity()
```

Comment:
If it's a city center, this is definitely a cityonfo - you can !! it and make the rest of the functions not with ?, And get rid of the yukky .let